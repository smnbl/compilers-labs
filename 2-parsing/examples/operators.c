int operators()
{
    int exponentiation_is_right_associative = 1 ^ 2 ^ 3;
    int addition_is_left_associative = 1 + 2 + 3;

    int precedence = foo = bar = 1 * -2 ^ 3 - 4 == 5 < 6;
}
