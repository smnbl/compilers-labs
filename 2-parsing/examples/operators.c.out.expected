└── Program
    └── FuncDecl: returnType = 'int', name = 'operators'
        └── CompoundStmt
            ├── VarDecl: type = 'int', name = 'exponentiation_is_right_associative'
            │   └── BinaryOpExpr: op = '^'
            │       ├── IntLiteral: value = '1'
            │       └── BinaryOpExpr: op = '^'
            │           ├── IntLiteral: value = '2'
            │           └── IntLiteral: value = '3'
            ├── VarDecl: type = 'int', name = 'addition_is_left_associative'
            │   └── BinaryOpExpr: op = '+'
            │       ├── BinaryOpExpr: op = '+'
            │       │   ├── IntLiteral: value = '1'
            │       │   └── IntLiteral: value = '2'
            │       └── IntLiteral: value = '3'
            └── VarDecl: type = 'int', name = 'precedence'
                └── BinaryOpExpr: op = '='
                    ├── VarRefExpr: name = 'foo'
                    └── BinaryOpExpr: op = '='
                        ├── VarRefExpr: name = 'bar'
                        └── BinaryOpExpr: op = '=='
                            ├── BinaryOpExpr: op = '-'
                            │   ├── BinaryOpExpr: op = '*'
                            │   │   ├── IntLiteral: value = '1'
                            │   │   └── UnaryOpExpr: op = '-'
                            │   │       └── BinaryOpExpr: op = '^'
                            │   │           ├── IntLiteral: value = '2'
                            │   │           └── IntLiteral: value = '3'
                            │   └── IntLiteral: value = '4'
                            └── BinaryOpExpr: op = '<'
                                ├── IntLiteral: value = '5'
                                └── IntLiteral: value = '6'
